pkg/smimetrics/client.go

```diff
@@ -143,23 +152,39 @@ func getMetricsResponse(k8sAPI *k8s.KubernetesAPI, path string, params map[strin
 }
 
 func parseTrafficMetrics(bytes []byte) (*TrafficMetrics, error) {
+	var status Status
+	json.Unmarshal(bytes, &status)
+	if status.Kind == "Status" {
+		return nil, errors.New(status.Message)
+	}
+
```

Comment:
Seems this checking logic can be factred out in its own method

Reply:
Thanks to go's verbose error checking, refactoring this logic into its own method actually net *added* 7 lines of code 🙃